src/matcha_ml/storage/azure_storage.py

```diff
@@ -88,19 +99,27 @@ def download_file(self, blob_client: BlobClient, dest_file: str) -> None:
             blob_data.readinto(my_blob)
 
     def download_folder(self, container_name: str, dest_folder_path: str) -> None:
-        """Download a folder from Azure Storage Container.
+        """Downloads a folder from Azure Storage Container.
+
+        Local directory is cleared beforehand, ensuring that the local files match the remote files.
 
         Args:
             container_name (str): Azure storage container name
             dest_folder_path (str): Path to folder to download all the files
         """
+        # Clears the local directory by removing all files, ensuring that it exclusively contains the files retrieved from Azure remote storage
+        for file in glob.glob(os.path.join(dest_folder_path, "*")):
```

Comment:
Should be a fix for now but this will change once we start taking into consideration the `.terraform` file i.e. terraform cache. We don't want to be deleting it should, download should preserve what is there (and ignored by upload).

Reply:
Will that be a fix implemented together with [RPD-220](https://fuzzy-labs.atlassian.net/jira/software/projects/RPD/boards/4?selectedIssue=RPD-220) Handle terraform cache better?

Reply:
Yes, RPD-220 should handle this properly. For now as part of this ticket, it's ok I guess to delete entire folder.

Reply:
I have updated the ticket description for [[RPD-220](https://fuzzy-labs.atlassian.net/jira/software/projects/RPD/boards/4?selectedIssue=RPD-220) Handle terraform cache better]?